root/etc/e-smith/templates/etc/ssh/sshd_config/45AllowGroups2Sshd

```diff
@@ -0,0 +1,58 @@
+{
+    # Allow groups to sftp/ssh with different policies
+    #
+    # if $sssd{ShellOverrideStatus} is disabled we are in legacy (everybody is allowed with a shell access)
+    # if $sssd{ShellOverrideStatus} is enabled we allow to root and members of domain admins group
+    # if $sssd{ShellOverrideStatus} is enabled  with non empty $sshd{AllowGroups} 
+    #   we allow root, members of domain admins group and members of groups inside the AllowGroups prop
+    #
+    # members restricted to sftp (SftpRestrictedGroups) must be allowed to sshd also
+
+    use strict;
+    use esmith::ConfigDB;
```

Comment:
This line can be removed and all the `$db->` calls below can be refactored with a simpler construct. The e-smith template system already makes the ConfigDB keys accessible through Perl variables, e.g. `$db->get_prop('sshd','PermitRootLogin')`  becomes `$sshd{'PermitRootLogin'}` and so on... The resulting code would be shorter and easier to read.

```suggestion
```

Reply:
done